rpc/cloudagent/network/networksecuritygroup/moc_cloudagent_networksecuritygroup.proto

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+syntax = "proto3";
+option go_package = "github.com/microsoft/moc/rpc/cloudagent/network";
+package moc.cloudagent.network;
+
+import "google/protobuf/wrappers.proto";
+import "moc_common_common.proto";
+import "moc_common_networkcommon.proto";
+
+enum Action {
+	Allow = 0;
+	Deny = 1;
+}
+
+enum Direction {
+	Inbound = 0;
+	Outbound = 1;
+}
+
+message NetworkSecurityGroupRequest {
+	repeated NetworkSecurityGroup NetworkSecurityGroups = 1;
+	Operation OperationType = 2;
+}
+
+message NetworkSecurityGroupResponse {
+	repeated NetworkSecurityGroup NetworkSecurityGroups = 1;
+	google.protobuf.BoolValue Result = 2;
+	string  Error = 3;
+}
+
+message NetworkSecurityGroupRule {
+	string name = 1;
+	string description = 2;
+	Action action = 3;
+	Direction direction = 4;
+	string sourceAddressPrefix = 5;
```

Comment:
Why do we have `sourceAddressPrefix` and `sourceAddressPrefixes`? similarly `destinationAddressPrefixes`, `sourcePortRanges` and `destinationPortRanges`?

Reply:
This was set up to match the API of the azure NSGs that we're trying to match, but we're not using the repeated ones in practice - because sourceAddressPrefix or sourcePortRange can be a comma separated list of address prefixes or port ranges, we just reuse that when needed. The repeated versions can be removed if needed

Reply:
We need to discuss and decide whether we need the sourceAddressPrefixes or not.